external-builds/pytorch/build_prod_wheels.py

```diff
@@ -600,11 +601,41 @@ def do_build_pytorch(
 ):
     # Compute version.
     pytorch_build_version = (pytorch_dir / "version.txt").read_text().strip()
+    pytorch_ver = pytorch_build_version
     pytorch_build_version += args.version_suffix
     print(f"  Default PYTORCH_BUILD_VERSION: {pytorch_build_version}")
-    print(
-        f"  Flash attention enabled: {args.enable_pytorch_flash_attention_windows or not is_windows}"
-    )
+
+    # Extract the major.minor version as a float (e.g., 2.10)
+    match = re.match(r"^(\d+)\.(\d+)", pytorch_ver)
+    if not match:
+        raise ValueError(f"Could not parse version from {pytorch_build_version}")
+    pytorch_version = float(f"{major}.{minor}")
+
+    ## Disable only for Linux on 2.10 Pytorch version
+    ## https://github.com/ROCm/TheRock/issues/1619
+    if not is_windows:
+        if pytorch_version < 2.10:
+            env["USE_FBGEMM_GENAI"] = "ON"
+            use_flash_attention = (
+                "1" if args.enable_pytorch_flash_attention_linux else "0"
+            )
+            env.update(
+                {
+                    "USE_FLASH_ATTENTION": use_flash_attention,
+                    "USE_MEM_EFF_ATTENTION": use_flash_attention,
+                }
+            )
+            print("FBGEMM_GENAI and Flash Attention enabled (PyTorch < 2.10, Linux)")
```

Comment:
The control flow and comments here are still confusing, especially once we re-enable support for either of these features. Maybe add separate sections for `USE_FBGEMM_GENAI` and `USE_FLASH_ATTENTION` so they aren't tightly coupled?